### DISCUSSION TITLE
Trying to extract both annotations and text from a PDF.

### DISCUSSION THREAD

**Discussion by Santoshzzz** [jsvine/pdfplumber#342] [Ask for help with specific PDFs]
Hi,
I'm trying to extract both annotations and text from a PDF. I'm using "text = p0.extract_words(x_tolerance=3, y_tolerance=3, keep_blank_chars=False, use_text_flow=False, horizontal_ltr=True, vertical_ttb=True, extra_attrs=[])
print(text)" to extract text but unable to extract the annotations in the same way.
Any help?

**Comment by jsvine** [comment]
PDFs represent annotations in a distinctly different way than standard text. You can access them via `my_page.annots` but, as such, are represented differently. See Section 8.4 of the [official PDF specification](https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/pdf_reference_1-7.pdf) for details.

**Reply by Santoshzzz** [reply]
Thank you. Can you help me with how can I extract them with the page numbers in the same program? Because I want to transform this into a dataframe and merge both text and annotation extracts.

**Reply by jsvine** [reply]
Each annotation's page number is embedded within the annotation object itself. E.g.,:

```python
for annot in pdf.annots:
    print(annot["page_number"])
```
